helix-term/src/application.rs

```diff
@@ -67,26 +67,24 @@ impl Application {
             std::sync::Arc::new(theme::Loader::new(&conf_dir, &helix_loader::runtime_dir()));
 
         let true_color = config.editor.true_color || crate::true_color();
-        let theme = config
-            .theme
-            .as_ref()
-            .and_then(|theme| {
-                theme_loader
-                    .load(theme)
+
+        let theme = match config.theme.as_ref() {
+            Some(theme_name) => {
+                match theme_loader
+                    .load(theme_name)
                     .map_err(|e| {
-                        log::warn!("failed to load theme `{}` - {}", theme, e);
+                        log::warn!("failed to load theme `{}` - {}", theme_name, e);
                         e
                     })
                     .ok()
                     .filter(|theme| (true_color || theme.is_16_color()))
-            })
-            .unwrap_or_else(|| {
-                if true_color {
-                    theme_loader.default()
-                } else {
-                    theme_loader.base16_default()
+                {
+                    Some(theme) => theme,
+                    None => Application::default_theme(theme_loader.clone(), true_color),
                 }
-            });
+            }
+            None => Application::default_theme(theme_loader.clone(), true_color),
+        };
```

Comment:
I think this part can be refactored. I think for this we can just map then unwrap_or_default so we don't need to do the whole block.

1. map user specified theme
2. unwrap_or_default default_theme